src/main/java/io/vertx/redis/client/Redis.java

```diff
@@ -59,24 +61,37 @@ static Redis createClient(Vertx vertx, String connectionString) {
    * @return the client
    */
   static Redis createClient(Vertx vertx, RedisOptions options) {
+    return createClient(vertx, options, () -> Future.succeededFuture(new RedisOptions(options)));
+  }
+
+  /**
+   * Create a new redis client using the given client options.
+   *
+   * @param vertx the vertx instance
+   * @param options the user provided options
+   * @param optionsSupplier the user provided options, which can be changed dynamically.
+   *                        Values like type cannot be changed.
+   * @return the client
+   */
+  @GenIgnore
```

Comment:
I was forced to add GenIgnore here. The generator could not handle the Supplier<Future<>> construct. Not sure if there is another way.

Reply:
you can try using instead `@GenIgnore(GenIgnore.PERMITTED_TYPE)` instead, we should support this better in the future indeed